From 95379c62a47518902c8b42d64cebe05c71fa6545 Mon Sep 17 00:00:00 2001
From: Yi-Shum <EasonChen1@quantatw.com>
Date: Fri, 21 Jun 2024 13:39:44 +0800
Subject: [PATCH 1/1] PwmSensor: Fix function getValue returns wrong data type

Summary:
Function "getValue" is defined as uint32_t but returns -1 in some cases.
Change it to return 0.

Tested:
- Build: PASS

Change-Id: Ifd228c94e4ebe14d9bdea953506d17371f9e5f79
Signed-off-by: Yi-Shum <EasonChen1@quantatw.com>
---
 src/PwmSensor.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/PwmSensor.cpp b/src/PwmSensor.cpp
index 2a431eb..43df6a2 100644
--- a/src/PwmSensor.cpp
+++ b/src/PwmSensor.cpp
@@ -211,12 +211,14 @@ uint32_t PwmSensor::getValue(bool errThrow)
     std::ifstream ref(sysPath);
     if (!ref.good())
     {
-        return -1;
+        std::cerr << "Error opening " << sysPath << "\n";
+        return 0;
     }
     std::string line;
     if (!std::getline(ref, line))
     {
-        return -1;
+        std::cerr << "Error reading pwm at " << sysPath << "\n";
+        return 0;
     }
     try
     {
@@ -225,7 +227,7 @@ uint32_t PwmSensor::getValue(bool errThrow)
     }
     catch (const std::invalid_argument&)
     {
-        std::cerr << "Error reading pwm at " << sysPath << "\n";
+        std::cerr << "Error converting pwm\n";
         // throw if not initial read to be caught by dbus bindings
         if (errThrow)
         {
-- 
2.34.1

